src/NHibernate.Test/TypesTest/DateTypeTest.cs

```diff
@@ -55,36 +56,41 @@ public void ReadWriteNormal()
 		{
 			var expected = DateTime.Today.Date;
 
-			var basic = new DateClass { DateValue = expected.AddHours(1) };
-			object savedId;
-			using (ISession s = OpenSession())
-			{
-				savedId = s.Save(basic);
-				s.Flush();
-			}
-			using (ISession s = OpenSession())
+			ReadWrite(expected);
+		}
+
+		[Test]
+		public void ReadWriteMin()
+		{
+			var expected = Sfi.ConnectionProvider.Driver.MinDate;
+
+			ReadWrite(expected);
+		}
+
+		[Test]
+		public void ReadWriteYear750()
+		{
+			var expected = new DateTime(750, 5, 13);
+			if (Sfi.ConnectionProvider.Driver.MinDate < expected)
```

Comment:
Shouldn't it be reversed?

Reply:
Yes, you are correct

Reply:
I am going to do some changes, I will do it by the way.